torture.c

```diff
@@ -8,18 +8,72 @@
 #include <stdlib.h>
 #include <time.h>
 #include <unistd.h>
+#include <string.h>
 #include <math.h>
 
+char buffer[1024];
+size_t buffer_pos = 0;
+
+const char help[] =
+    "Usage: %s [-h] [-2] [-c] [-g] [-r rate]\n"
+    "  -h       print this help message and exit\n"
+    "  -2       output two waves\n"
+    "  -c       randomly chunk the output\n"
+    "  -g       occasionally output garbage\n"
+    "  -r rate  sample rate in samples/s (default: 100)\n";
+
+const char optstring[] = "h2cgr:";
+
 int main(int argc, char *argv[]) {
-    (void) argv;
-    unsigned int n;
-
-    for(n=0;;n+=5) {
-        printf("%.1f\n", (sin(n*M_PI/180)*5)+5);
-        if(argc==2)
-            printf("%.1f\n", (cos(n*M_PI/180)*5)+5);
-        fflush(stdout);
-        usleep(10000);
+    int opt;
+    int two_waves = 0;
+    int chunked = 0;
+    int add_garbage = 0;
+    double rate = 100;
+
+    // Parse the command line.
+    while ((opt = getopt(argc, argv, optstring)) != -1) {
+        switch (opt) {
+            case 'h': printf(help, argv[0]); return EXIT_SUCCESS;
+            case '2': two_waves = 1; break;
+            case 'c': chunked = 1; break;
+            case 'g': add_garbage = 1; break;
+            case 'r': rate = atof(optarg); break;
+            default: fprintf(stderr, help, argv[0]); return EXIT_FAILURE;
+        }
+    }
+    if (argc > optind) {
+        fprintf(stderr, help, argv[0]);
+        return EXIT_FAILURE;
+    }
+
+    useconds_t delay = 1e6 / rate;
+
+    for (unsigned int n=0; ; n+=5) {
+        buffer_pos += sprintf(buffer + buffer_pos, "%.1f ", (sin(n*M_PI/180)*5)+5);
+        if (add_garbage && rand() <= RAND_MAX / 5)
```

Comment:
Since the new code calls `rand`, please also call `srand` with a value that can be overridden through a new CLI argument — e.g. `-s seed` — so that  deterministic operation remains possible when needed. Thank you!